packages/styled-components/components/pagination/index.js

```diff
@@ -0,0 +1,223 @@
+/* eslint max-len: 0 */
+import React from 'react';
+import PropTypes from 'prop-types';
+import queryString from 'query-string';
+import withThemes from '@irvingjs/styled/components/withThemes';
+import Link from 'components/link';
+import * as defaultStyles from './themes/default';
+
+/**
+ * Using the same props from a Pagination component, build the url for a given
+ * page.
+ *
+ * @param {object}  props Pagination component props.
+ * @param {integer} page  Page for the url.
+ * @return {string}
+ */
+const buildUrl = (props, page) => {
+  const {
+    baseUrl,
+    paginationFormat,
+  } = props;
+
+  const paginationPart = paginationFormat.replace('%1$d', page);
+  const currentQueryVars = queryString.parse(window.location.search);
+
+  if (1 === page) {
+    return `${baseUrl}?${queryString.stringify(currentQueryVars)}`;
+  }
+
+  return `${baseUrl}${paginationPart}?${queryString.stringify(currentQueryVars)}`;
+};
+
+/**
+ * Pagination.
+ *
+ * Pagination UI.
+ *
+ * @todo Setup I18N.
+ * @todo Write tests.
+ */
+const Pagination = (props) => {
+  const {
+    // eslint-disable-next-line no-unused-vars
+    baseUrl,
+    currentPage,
+    displayFirstAndLast,
+    displayPrevAndNext,
+    // eslint-disable-next-line no-unused-vars
+    paginationFormat,
```

Comment:
Doing this and the other `disable` so that we can still validate the type + get a default, but they're only used in the context of `buildUrl()`.